src/openfl/display/SubBitmapData.hx

```diff
@@ -0,0 +1,310 @@
+package openfl.display;
+
+import openfl.display.BitmapData;
+
+#if !display
+import js.Browser;
+import js.html.CanvasElement;
+import openfl._internal.renderer.RenderSession;
+import openfl.geom.ColorTransform;
+import lime.graphics.GLRenderContext;
+import lime.graphics.opengl.GL;
+import lime.graphics.opengl.GLBuffer;
+import lime.graphics.opengl.GLTexture;
+import lime.graphics.opengl.WebGLContext;
+import lime.graphics.utils.ImageCanvasUtil;
+import lime.utils.Float32Array;
+#end
+
+@:access(openfl.geom.ColorTransform)
+class SubBitmapData extends BitmapData {
+	private var __offsetX = 0;
+	private var __offsetY = 0;
+	private var __texX0 = 0.0;
+	private var __texX1 = 1.0;
+	private var __texY0 = 0.0;
+	private var __texY1 = 1.0;
+	private var __texX:Int;
+	private var __texY:Int;
+	private var __texWidth:Int;
+	private var __texHeight:Int;
+	private var __parentBitmap:BitmapData;
+	private var __rotated = false;
+
+	public function new(atlasBitmap:BitmapData, x:Int, y:Int, width:Int, height:Int, frameX:Int, frameY:Int, frameWidth:Int, frameHeight:Int, rotated:Bool) {
+		#if !display
+		var w = width, h = height;
+		var offsetX = 0;
+		var offsetY = 0;
+		if (frameWidth > 0 && frameHeight > 0) {
+			offsetX = -frameX;
+			offsetY = -frameY;
+			w = frameWidth;
+			h = frameHeight;
+		} else if (rotated) {
```

Comment:
It could have frame and be rotated right ?

Reply:
When there's frame, the frameWidth and frameHeight are already "rotated"